File: Bascis of Java/Nth Fibonacci number.java
public class Solution{
   public static int fibonacciNumber(int n){
     int m=1000000007;
     int first=0,second=1;
       int fibo=1;
       if(n==0) return 0;
       else if(n==1) return 1;
       for(int i=2;i<=n;i++){
           fibo=(first+second)%m;
           first=second;
           second=fibo;
       }
       return fibo;
   }
}

File: Basics of C++/N-th Fibonacci Number.cpp
int Lnumber[1000000];

int fibonacciNumber(int n){
if(n<=1)
return n;
   int x,y;
    
if(Lnumber[n-1]!=0){
      x=Lnumber[n-1]%1000000007;
   }
   else{
       x=fibonacciNumber(n-1);
   }
if(Lnumber[n-2]!=0){
       y=Lnumber[n-2]%1000000007;
   }
   else{
       y=fibonacciNumber(n-2);
   }
return Lnumber[n]=(x+y)%1000000007;
}